lms/services/d2l_api/client.py

```diff
@@ -193,3 +196,17 @@ def _find_files(self, course_id, modules):
                 "updated_at": module["updated_at"],
                 "children": module_files + list(module_children),
             }
+
+    def _files_for_storage(self, course_id, files, parent_id=None):
+        for file in files:
+            yield {
+                "type": "d2l_file" if file["type"] == "File" else "d2l_folder",
+                "course_id": course_id,
+                "lms_id": file["id"],
+                "name": file["display_name"],
+                "parent_lms_id": parent_id,
+            }
+
+            yield from self._files_for_storage(
+                course_id, file.get("children", []), file["id"]
+            )
```

Comment:
Presumably the two formats here are:

 * The format expected by the front end
 * The format expected by the DB / file service
 
It's a shame they are different, but they are. 

I wonder if it's worth extracting this conversion between formats and flattening behavior into something re-usable?  Then you could emit the tree with children in one function, and then have it flatten and reformat itself as required? This could presumably be used elsewhere (Canvas / Blackboard?), where we really should be doing the same thing.

This might also nudge us towards emitting the correct things at the service level, rather than our usual of emitting something in an LMS specific format and then having to convert it a lot.